internal/app/networkd/pkg/nic/nic.go

```diff
@@ -196,6 +254,26 @@ func (n *NetworkInterface) Addressing() error {
 	return nil
 }
 
+// AddressingSub handles the address method for a configured sub interface ( dhcp/static ).
+// This is inclusive of the address itself as well as any defined routes.
+func (n *NetworkInterface) AddressingSub() error {
+	if n.IsIgnored() {
+		return nil
+	}
+
+	for _, vlan := range n.Vlans {
+		for _, method := range vlan.AddressMethod {
+			if err := n.configureInterface(method, vlan.Link); err != nil {
+				// Treat as non fatal error when failing to configure an interface
```

Comment:
should we still log the errors for debugging?

Reply:
Added a log for this